docs/trace/data-prepper-reference.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: default
+title: Configuration Reference
+parent: Trace Analytics
+nav_order: 25
+---
+
+# Data Prepper configuration reference
+
+This page lists all supported Data Prepper sources, buffers, processors, and sinks, along with their associated options.
+
+
+## Sources
+
+Sources define where your data comes from.
+
+
+### otel_trace_source
+
+Source for the OpenTelemetry Collector.
+
+Option | Required | Description
+:--- | :--- | :---
+ssl | No | Boolean, whether to connect to the OpenTelemetry Collector over SSL.
+sslKeyCertChainFile | No | String, path to the security certificate (e.g. `"config/demo-data-prepper.crt"`.
+sslKeyFile | No | String, path to the security certificate key (e.g. `"config/demo-data-prepper.key"`).
+
+
+### file
+
+Source for flat file input.
+
+Option | Required | Description
+:--- | :--- | :---
+path | Yes | String, path to the input file (e.g. `logs/my-log.log`).
+
+
+### pipeline
+
+Source for reading from another pipeline.
+
+Option | Required | Description
+:--- | :--- | :---
+name | Yes | String, name of the pipeline to read from.
+
+
+### stdin
+
+Source for console input. Can be useful for testing. No options.
+
+
+## Buffers
+
+Buffers store data as it passes through the pipeline. If you implement a custom buffer, it can be memory-based (better performance) or disk-based (larger).
+
+
+### bounded_blocking
+
+The default buffer. Memory-based.
+
+Option | Required | Description
+:--- | :--- | :---
+buffer_size | No | Integer, the maximum number of records the buffer accepts.
```

Comment:
Should we add details about default values if option is not provided by the user

Reply:
Sure, added. Defaults here appear to be 8 and 512:

```
private static final int DEFAULT_BUFFER_CAPACITY = 512;
private static final int DEFAULT_BATCH_SIZE = 8;
```